src/Microsoft.AspNetCore.OData/Extensions/ODataRouteBuilderExtensions.cs

```diff
@@ -25,6 +25,382 @@ namespace Microsoft.AspNet.OData.Extensions
     /// </summary>
     public static class ODataRouteBuilderExtensions
     {
+        /// <summary>
+        /// Sets the <see cref="DefaultQuerySettings"/> in route builder.
+        /// </summary>
+        /// <param name="builder">The <see cref="IRouteBuilder"/>.</param>
+        /// <param name="defaultQuerySettings">The default query settings.</param>
+        public static IRouteBuilder SetDefaultQuerySettings(this IRouteBuilder builder, DefaultQuerySettings defaultQuerySettings)
+        {
+            if (builder == null)
+            {
+                throw Error.ArgumentNull("builder");
+            }
+
+            if (defaultQuerySettings == null)
+            {
+                throw Error.ArgumentNull("defaultQuerySettings");
+            }
+
+            if (!defaultQuerySettings.MaxTop.HasValue || defaultQuerySettings.MaxTop > 0)
+            {
+                ModelBoundQuerySettings.DefaultModelBoundQuerySettings.MaxTop = defaultQuerySettings.MaxTop;
```

Comment:
Double-checking, we're setting the `ModelBoundQuerySettings.DefaultModelBoundQuerySettings.MaxTop` if `defaultQuerySettings` doesn't have a max top value or when it's greater than 0?

When `defaultQuerySettings.MaxTop` doesn't have a value, does it have a default and we're setting `ModelBoundQuerySettings.DefaultModelBoundQuerySettings.MaxTop` to it? If that's the case, is the default value 0?

Reply:
I'm wondering if you mean `&&` instead of `||`.

Reply:
It's the same as the code for AspNet in HttpConfigurationExtensions.cs, basically if you have a non-0 value (null or > 0), use that instead. Yes, 0 is default.